enhancements/network/ovs-hardware-offload.md

```diff
@@ -0,0 +1,242 @@
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+- When will RHCOS uses RHEL-8.3 as base image?
+- Any known issues with connction track feature using OVS Hardware Offload?
+- Any known issues with IPv6 traffic using OVS Hardware Offload?
+
+## Summary
+
+This enhancement is to introduce OVS hardware offload feature in OpenShift
+for high-performance datapath with OVN-Kubernetes. OVS hardware offload
+enables offloading of OVS datapath rules to a compatible network card using
+linux Traffic Control (TC) tool `tc`. It relies on SR-IOV technology to control
+the SR-IOV Virtual Functions (VF) via VF representor port that is plugged
+into OpenvSwitch bridge. It improves the utilization of available bandwidth
+on underlying NICs by moving CPU intensive network packet processing from
+OVS software to hardware.
+
+## Motivation
+
+Enable high-performance datapath on primary pod interface with production-ready
+OVN-Kubernetes.
+
+### Goals
+
+Enable OpenFlow hardware offload for OVN datapath
+
+### Non-Goals
+
+Replace OpenFlow software based solution
+
+## Proposal
+
+OVS Hardware Offload accelerates the OVS software flows with a SR-IOV NIC
+that works in `switchdev` mode. It requires configurations in both OpenvSwitch
+daemon and SR-IOV hardware NIC.
+
+- OpenvSwitch daemon is managed by Cluster Network Operator (CNO) and runs in
+`ovs-node` DaemonSet. A boolean configurable field will be added in CNO
+`Network` Custom Resource Definition `spec.defaultNetwork.ovnKubernetesConfig`
+to enable or disable OVS Hardware Offload feature for OpenvSwitch daemon.
+
+- SR-IOV hardware NIC is managed by SR-IOV Network Operator and will be
+provisioned with a new device type `switchdev`. The initial supported NIC
+model will be Mellanox MT27800 Family [ConnectX-5] 25GbE dual-port SFP28.
+
+Besides CNO and SR-IOV Operator changes, there are several other dependencies.
+
+- OVS Hardware Offload will only be supported on bare metal deployment.
+
+- Multus supports overwriting cluster network definition by specifying a
+customized default net-attach-def in pod annotation. Customized net-attach-def
+will still use `ovn-k8s-cni-overlay` CNI plugin, but with an special annotation
+that requests a SR-IOV device.
+
+- RHCOS image shall be based on RHEL-8.3 for fully support of OVS Hardware
+Offload with connection track feature.
+
+- OpenvSwitch package version 2.13.
+
+- Mellanox SR-IOV VF-LAG provides hardware link-aggregation (LAG) for hardware
+offloaded ports. It allows offloading of OVS rules to a linux `bond` network
+device that combines two idential and hardware offload capable Physical
+Functions (PFs). In order to use SR-IOV VF-LAG, a bond interface needs to be
```

Comment:
It's not clear if you're describing a required implementation detail of OVS offload ("you have to create a `bond` device even if you aren't using bonding") or if you're describing how OVS offload and bonding interact with each other. ("If you're using bonding and want OVS offload too, then you have to set things up like THIS, rather than like THAT.")

Reply:
Updated, please take a second look on the update.
It now starts with `To use OVS Hardware Offload with link redundency, a bond interface needs to
be created and used as geneve endpoint of OVS bridge.`